src/msw/listctrl.cpp

```diff
@@ -1780,37 +1774,41 @@ wxListCtrl::HitTest(const wxPoint& point, int& flags, long *ptrSubItem) const
 
     flags = 0;
 
-    if ( hitTestInfo.flags & LVHT_ABOVE )
-        flags |= wxLIST_HITTEST_ABOVE;
-    if ( hitTestInfo.flags & LVHT_BELOW )
-        flags |= wxLIST_HITTEST_BELOW;
+    // test whether an actual item was hit or not
+    if ( hitTestInfo.flags == LVHT_ONITEMICON )
+        flags = wxLIST_HITTEST_ONITEMICON;
+    // note a bug in comtl32.dll:
+    // 1) in report mode, when there are 2 or more columns, if you click to the
+    //    right of the first column, ListView_HitTest() returns LVHT_ONITEM
+    //    (LVHT_ONITEMICON|LVHT_ONITEMLABEL|LVHT_ONITEMSTATEICON).
+    else if ( hitTestInfo.flags == LVHT_ONITEMLABEL ||
+              (hitTestInfo.flags == LVHT_ONITEM && InReportView()) )
+        flags = wxLIST_HITTEST_ONITEMLABEL;
```

Comment:
This cascaded `if` would be more readable if we put braces around the statement bodies, especially this one where the condition itself is on multiple lines (but then all the other ones too, for consistency).